libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -290,9 +290,60 @@ void ArrayUtils::copyArrayToStorage(ArrayType const& _targetType, ArrayType cons
 			_context << Instruction::POP << Instruction::SWAP1 << Instruction::POP;
 			// stack: target_ref target_data_end target_data_pos_updated
 			if (targetBaseType->storageBytes() < 32)
-				utils.clearStorageLoop(TypeProvider::uint256());
+			{
+				if (!targetType->isDynamicallySized() && !sourceType->isDynamicallySized())
+					if (auto const* targetArrayType = dynamic_cast<ArrayType const*>(targetType))
+					{
+						auto const* sourceArrayType = dynamic_cast<ArrayType const*>(sourceType);
+						solAssert(sourceArrayType);
```

Comment:
Is this bit added here because it is not guaranteed that when there's nothing to clear the position is equal to the end? If so, it needs a comment providing that context. Otherwise it looks just like an optimization (which could be done just as well in a separate PR).